SF iOS/SF iOS/Feed/EventsFeedViewController.m

```diff
@@ -175,12 +242,39 @@ - (void)previewingContext:(id<UIViewControllerPreviewing>)previewingContext comm
     [self presentViewController:viewControllerToCommit animated:true completion:nil];
 }
 
+//MARK: - UISearchBarDelegate
+
+- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
+    [self.dataSource filterEventsWith:searchText];
+}
+
+- (void)searchBarTextDidEndEditing:(UISearchBar *)searchBar {
+    [searchBar resignFirstResponder];
+    [self.dataSource filterEventsWith:searchBar.text];
+}
+
+- (void)searchBarCancelButtonClicked:(UISearchBar *)searchBar {
+    [searchBar resignFirstResponder];
+    [self.dataSource filterEventsWith:@""];
+    searchBar.text = @"";
+    [self.tableView scrollToRowAtIndexPath:[NSIndexPath indexPathForRow:0 inSection:0] atScrollPosition:UITableViewScrollPositionTop animated:YES];
```

Comment:
similar thought here; what if you've scrolled a bunch and tap on the search bar, but dismiss it before typing anything? do we want to lose the scroll position?